src/lib/helpers/config.ts

```diff
@@ -0,0 +1,65 @@
+import { z, ZodError } from "https://deno.land/x/zod@v3.24.2/mod.ts";
+import { parse } from "jsr:@std/toml";
+
+const ConfigSchema = z.object({
+    server: z.object({
+        port: z.number().default(8282),
+        host: z.string().default("127.0.0.1"),
+        secret_key: z.string().default("CHANGE_ME"),
+        verify_requests: z.boolean().default(false),
+    }).strict().default({}),
+    cache: z.object({
+        enabled: z.boolean().default(true),
+        directory: z.string().default("/var/tmp"),
+    }).strict().default({}),
+    networking: z.object({
+        ump: z.boolean().default(false),
+        proxy: z.string().optional(),
+        fetch: z.object({
+            timeout_ms: z.number().optional(),
+            retry: z.object({
+                enabled: z.boolean(),
+                times: z.number().optional(),
+                initial_debounce: z.number().optional(),
+                debounce_multiplier: z.number().optional(),
+            }).strict().optional(),
+        }).strict().optional(),
+    }).strict().default({}),
+    jobs: z.object({
+        youtube_session: z.object({
+            po_token_enabled: z.boolean().default(true),
+            frequency: z.string().default("*/5 * * * *"),
+        }).strict().default({}),
+    }).strict().default({}),
+    youtube_session: z.object({
+        oauth_enabled: z.boolean().default(false),
+        cookies: z.string().default(""),
+    }).strict().default({}),
+}).strict();
+
+export type Config = z.infer<typeof ConfigSchema>;
+
+export async function parseConfig() {
+    try {
+        const configFileContents = await Deno.readTextFile("config/local.toml").catch(() => null);
```

Comment:
What do you think of renaming the file to like config/config.toml? And we rename default.toml to config.example.toml?

This way, we do like on Invidious. We comment each parameter and people have the flexibility of enabling which one they prefer by renaming config.example.toml to config.toml

It's ok to have a new filename, it's still in testing. And I much rather do that right now than once it's "full" in production/stable branch of invidious.

Reply:
yep, absolutely agree. `local.toml` doesn't make sense if `default.toml` goes.

Reply:
OK for me to do that change in this Pr then :)